lib/em-http/http_connection_options.rb

```diff
@@ -42,4 +42,19 @@ def connect_proxy?
   def socks_proxy?
     @proxy && (@proxy[:type] == :socks5)
   end
+
+  def proxy
+    @proxy ||= proxy_env_uri && { :host => proxy_env_uri.host, :port => proxy_env_uri.port, :type => :http }
```

Comment:
> I'm wondering if we can unroll and make this a bit more explicit..

No problem;  I was trying to avoid over-loading the class with too many new methods.

> ` @proxy = options[:proxy] || env_proxy`

I do tend to follow the `def proxy; @proxy ||= ...; end` style, but I take your point that this is probably an inappropriate place to use it.
I'll rework.